flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/ClusterPartitionDescriptorImpl.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.executiongraph;
+
+import org.apache.flink.api.common.ClusterPartitionDescriptor;
+import org.apache.flink.runtime.shuffle.ShuffleDescriptor;
+
+public class ClusterPartitionDescriptorImpl implements ClusterPartitionDescriptor {
+    private ShuffleDescriptor shuffleDescriptor;
+	private final int numberOfSubpartitions;
+
+    public ClusterPartitionDescriptorImpl(ShuffleDescriptor shuffleDescriptor,
+										  int numberOfSubpartitions) {
+        this.shuffleDescriptor = shuffleDescriptor;
+		this.numberOfSubpartitions = numberOfSubpartitions;
+	}
+
+    public ShuffleDescriptor getShuffleDescriptor() {
+        return shuffleDescriptor;
+    }
+
+    public void setShuffleDescriptor(ShuffleDescriptor shuffleDescriptor) {
```

Comment:
Does this descriptor have to be mutable? In general we prefer the classes to be immutable if possible.